temporal-sdk/src/main/java/io/temporal/common/converter/DefaultDataConverter.java

```diff
@@ -82,6 +88,10 @@ public static void setDefaultDataConverter(DataConverter converter) {
   /**
    * Creates a new instance of {@code DefaultDataConverter} populated with the default list of
    * payload converters.
+   *
+   * <p>The newly created instance will use the default failure converter, configured to not encode
+   * message and stack trace. Call {@code withFailureConverterOverride} on the instance to set a
```

Comment:
This sends an incorrect message to users. I think we should actually remove this doc completely to don't give wrong ideas.
Instead, on CodecConverter we should provide information that it takes care of encoding failures if configured to do so with the right constructor flag.